### ISSUE TITLE
ability to limit the number of sh:ValidationResults

### ISSUE THREAD

**Issue by VladimirAlexiev** [w3c/data-shapes#161]
Usually validation errors are due to some systemic reason (eg an RDFization process has a mistake).
In that case you don't get a couple of ValidationResults for a given reason, you get thousands or millions.

It is more practical to have the ability to put some limit(s). Rdf4j has this ability: https://rdf4j.org/documentation/programming/shacl/#limiting-the-validation-report (total results, and results per constraint)

After the data producer fixes the root cause of errors in this limited report, they can rerun the validator, fix another batch of errors, etc.
This is similar to how many compilers abort after the encounter 1000 errors: the programmer is not likely to fix more than 1000; and because often one error precipitates multiple following errors, it's better to work in digestible chunks.

**Comment by HolgerKnublauch** [comment]
Yes, TopBraid has the same options. But this may be outside of the standardization process unless we define some sort of validation API similar to the SPARQL protocol.

**Comment by tfrancart** [comment]
In the same idea, having a summary of the validation report with the count of similar errors is very useful. In SHACL Play we do this:

![Image](https://github.com/user-attachments/assets/9d6d0c83-1bf3-40af-806b-7598ab4eae6c)

If I remember correctly, we group simply based on the message of the validation result (suboptimal).
This enables to fix the errors that occur the most often first.

**Comment by VladimirAlexiev** [comment]
The EU [EU Interoperability Test Bed (ITB)](https://github.com/ISAITB/shacl-validator) can validate [RDF data](https://www.itb.ec.europa.eu/shacl/any/upload) or [SHACL shapes](https://www.itb.ec.europa.eu/shacl/shacl/upload) and has a button "Summary report" that presents counts. However, it first always produces all individual results.

**Comment by HolgerKnublauch** [comment]
@VladimirAlexiev can this be closed? I have responded above that this sounds like outside of the standard unless we define a protocol (which is not scheduled).

**Comment by nicholascar** [comment]
@VladimirAlexiev I agree this is interesting to the SHACL community but also think it is outside Core, at the very least. I'm removing the Core label.

**Comment by TallTed** [comment]
I agree. This feels like an optional implementation detail. That said, I would like to have it in any implementation I use.

**Comment by HolgerKnublauch** [comment]
Closed due to inactivity and as discussed.
